File: env/env.go
package env

import (
	"os"
)

var Dev = os.Getenv("DEV") == "1"
var Prod = !Dev
var SecretToken string
var SigningKey string
var DbURL = os.Getenv("DATABASE_URL")
var DebugDB = os.Getenv("DB_DEBUG") == "1"

func init() {
	if Dev {
		SigningKey = "development-key"
	} else {
		SigningKey = os.Getenv("SIGNING_KEY")
		if SigningKey == "" {
			panic("Must provide a signing key in prod.")
		}
	}
	if Dev {
		SecretToken = "development-token"
	} else {
		SecretToken = os.Getenv("SECRET_TOKEN")
		if SecretToken == "" {
			panic("Must provide a secret token in prod.")
		}
	}
}

File: src/env.rs
use std::env;

use once_cell::sync::Lazy;

#[derive(Clone, Copy, Debug, Eq, PartialEq)]
pub enum Env {
    Dev,
    Prod,
}

impl Env {
    pub fn dev() -> bool {
        Self::default() == Self::Dev
    }
    pub fn prod() -> bool {
        Self::default() == Self::Prod
    }
}

impl Default for Env {
    fn default() -> Self {
        if &env::var("DEV").unwrap_or_else(|_| "0".into()) == "1" {
            Env::Dev
        } else {
            Env::Prod
        }
    }
}

pub static SIGNING_KEY: Lazy<String> = Lazy::new(|| {
    if Env::dev() {
        "development-key".into()
    } else {
        env::var("SIGNING_KEY").expect("Must provide a signing key in prod.")
    }
});

pub static SECRET_TOKEN: Lazy<String> = Lazy::new(|| {
    if Env::dev() {
        "development-token-00000000000000000000000000000000000000000000000".into()
    } else {
        env::var("SECRET_TOKEN").expect("Must provide a secret token in prod.")
    }
});
